app/views/providers/means_summaries/_income_cash_payments.html.erb

```diff
@@ -0,0 +1,54 @@
+<!-- TODO: share with means report?! -->
+<div class="govuk-grid-row">
+    <div class="govuk-grid-column-two-thirds">
+        <h3 class="govuk-heading-m"><%= t('.heading') %></h3>
+    </div>
+    <div class="govuk-grid-column-one-third govuk-summary-list--no-border align-text-right">
+        <p>
+            <%= link_to_accessible(t('generic.change'),
+                                   check_answer_url_for(:providers,
+                                                        :cash_incomes,
+                                                        @legal_aid_application),
+                                   class: 'govuk-link change-link', suffix: t('.heading')) %>
+        </p>
+    </div>
+</div>
+
+<%
+credits = @legal_aid_application.legal_aid_application_transaction_types.credits
+cash_credits = @legal_aid_application.cash_transactions.credits.order(:transaction_date)
+credit_categories = TransactionType.credits.not_children
+%>
+
+<% if !@legal_aid_application.transaction_types.credits.exists? %>
+ <div class="govuk-body>"> <%= t('generic.none') %> </div>
+<% end %>
+
+<dl id="income-cash-payments-questions" class="govuk-summary-list govuk-!-margin-bottom-9">
+  <% credit_categories.each do |category| %>
+    <% if credits.pluck(:transaction_type_id).include?(category.id) %>
+      <% if cash_credits.pluck(:transaction_type_id).include?(category.id) %>
+        <%
+        cash_payments = cash_credits.where(transaction_type_id: category.id)
+
+        cash_payment_list = cash_payments.each_with_object([]) do |payment, list|
+          list << "#{payment.transaction_date.strftime("%B %Y")} #{gds_number_to_currency(payment.amount, precision: 2)}"
+        end
+        %>
+
+        <%= check_answer_no_link(
+            name: category.label_name,
+            question: category.label_name,
+            answer: cash_payment_list,
+            read_only: read_only) %>
+
+      <% else %>
+        <%= check_answer_no_link(
+            name: category.label_name,
+            question: category.label_name,
+            answer: t('generic.none'),
+            read_only: read_only) %>
+      <% end %>
+    <% end %>
```

Comment:
This feels like a lot of complexity for a view, and both these partials feel like really good candidates for domain-specific [ViewComponents](https://viewcomponent.org).

I'll raise an example PR to start an asynchronous discussion with the rest of the team, but in the meantime, I wonder if it's worthwhile moving some of this logic to the underlying object/a presenter/helper methods?

Reply:
yep, I have started refactoring now.

Reply:
have refactored out but will look up view components and other presenter type possibilities in my L&D i think